libparsec/crates/platform_device_loader/src/native/mod.rs

```diff
@@ -233,11 +279,66 @@ pub async fn load_device(
     Ok(Arc::new(device))
 }
 
+fn save_device_file(key_file: &PathBuf, file_content: &DeviceFile) -> Result<(), SaveDeviceError> {
+    let file_content = file_content.dump();
+
+    if let Some(parent) = key_file.parent() {
+        std::fs::create_dir_all(parent).map_err(|e| SaveDeviceError::InvalidPath(e.into()))?;
+    }
+    let tmp_path = match key_file.file_name() {
+        Some(file_name) => {
+            let mut tmp_path = key_file.clone();
+            {
+                let mut tmp_file_name = file_name.to_owned();
+                tmp_file_name.push(".tmp");
+                tmp_path.set_file_name(tmp_file_name);
+            }
+            tmp_path
+        }
+        None => {
+            return Err(SaveDeviceError::InvalidPath(anyhow::anyhow!(
+                "Path is missing a file name"
+            )))
+        }
+    };
+
+    // Classic pattern for atomic file creation:
+    // - First write the file in a temporary location
+    // - Then move the file to it final location
+    // This way a crash during file write won't end up with a corrupted
+    // file in the final location.
+    std::fs::write(&tmp_path, file_content).map_err(|e| SaveDeviceError::InvalidPath(e.into()))?;
+    std::fs::rename(&tmp_path, key_file).map_err(|e| SaveDeviceError::InvalidPath(e.into()))?;
```

Comment:
I reused it and I/O with FS should be in this crate

Reply:
By "reused" I meant having this code inside a _function_ that can be reused.

Reply:
It is in a function :thinking:

Reply:
It can be generic if we take slice instead of DeviceFile, but for the moment it is used only with

Reply:
> It is in a function 🤔

Right, is in the function `save_device_file`. My question was if this could be used to save other files (not only device file). Anyway, that can be done later, so I will resolve this conversation.